### PR TITLE
[SPARK-50497][SQL] Fail queries with proper message if MultiAlias contains non-generator function

### PR DESCRIPTION
**Pull request by mihailomilosevic2001**
### What changes were proposed in this pull request?
MultiAlias fails if the underlying function is not a generator.

### Why are the changes needed?
Syntax for multi aliasing should return proper error and not fail in check analysis with unresolved.

### Does this PR introduce _any_ user-facing change?
Yes, error message changes

### How was this patch tested?
Added test.

### Was this patch authored or co-authored using generative AI tooling?
Yes.

### CODE REVIEW

**Comment by mihailom-db** [comment]
@MaxGekk please review this.

**Comment by mihailom-db** [comment]
@vladimirg-db for context.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala [hunk 1]
```diff
@@ -482,6 +482,10 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog with QueryErrorsB
               errorClass = "UNBOUND_SQL_PARAMETER",
               messageParameters = Map("name" -> p.name))
 
+          case ma@MultiAlias(child, names) if child.resolved && !child.isInstanceOf[Generator] =>
```
Comment (vladimirg-db) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala:485]: Why exactly do we check for `child.resolved` here?

Comment (mihailom-db) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala:485]: If child is unresolved, we cannot check for Generator, as we would usually have UnresolvedFunction.

File: common/utils/src/main/resources/error/error-conditions.json [hunk 2]
```diff
@@ -3574,6 +3574,12 @@
     ],
     "sqlState" : "42710"
   },
+  "MULTI_ALIAS_WITHOUT_GENERATOR" : {
+    "message" : [
+      "Multi part aliasing (<names>) is not supported with <expr> as it is not a generator function."
```
Comment (cloud-fan) [inline common/utils/src/main/resources/error/error-conditions.json:3579]: is `Multi part aliasing` an official name in SQL standard?

Comment (srielau) [reply common/utils/src/main/resources/error/error-conditions.json:3579]: It's not a concept at all. I don't have a better name.

**Review by srielau — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**